client/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+use ed25519_dalek::{Keypair, PublicKey};
+use failure::{err_msg, Error};
+use hpos_state_core::admin_keypair_from;
+use serde::*;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct HpAdminKeypair(Keypair);
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Payload {
+    method: String,
+    uri: String,
+    body_string: String,
+}
+
+#[wasm_bindgen]
+impl HpAdminKeypair {
+    /// @description Create an Ed25519 keypair for API signing calls to HPOS
+    /// @example
+    /// myKeys = new HpAdminKeypair( hc_public_key_string, email, password );
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        hc_public_key_string: String,
+        email: String,
+        password: String,
+    ) -> Result<HpAdminKeypair, JsValue> {
+        // rewrite any Error into JsValue
+        match new_inner(hc_public_key_string, email, password) {
```

Comment:
I know some of our older WASM code uses inner/outer split to handle errors properly. Have you tried `into_js_error` function, as seen in Chaperone `KeyManager`? I think it'd be much easier to read.

Reply:
leaving as is